app/models/agenda-item.ts

```diff
@@ -8,9 +14,19 @@ export default class AgendaItemModel extends Model {
   @attr('string') declare alternateLink: string;
   @attr('boolean') declare plannedPublic: boolean;
 
-  @belongsTo('session', { async: true, inverse: 'agendaItems' })
-  declare session?: SessionModel;
+  @hasMany('session', { async: true, inverse: 'agendaItems' })
+  declare sessions?: AsyncHasMany<SessionModel>;
 
   @belongsTo('agenda-item-handling', { async: true, inverse: null })
-  declare handledBy?: AgendaItemHandlingModel;
+  declare handledBy?: AsyncBelongsTo<AgendaItemHandlingModel>;
+
+  /**
+   * @returns the first session with hasMunicipality == true
+   * This is the session we want to use to resolve municipality & governingBody name
+   */
+  get session() {
+    return this.sessions?.slice().find((session) => {
```

Comment:
Accessing async relationships in a sync way like this is deprecated. `.slice()` also triggers deprecations.

If the relationship needs to stay async we can use the [reference APIs](https://api.emberjs.com/ember-data/5.1/classes/Model/methods/hasMany?anchor=hasMany), assuming the relationship was already loaded (via includes for example.

```suggestion
    return this.hasMany('sessions').value()?.slice().find((session) => {
```

TS is complaining though, and I'm not sure what the issue is, I think it's another bug in the DT types.

Reply:
Did some digging and it seems to be this issue: https://github.com/typed-ember/ember-cli-typescript/issues/1416

The mentioned workaround in the last comment seems to work.
```suggestion
    return (this as AgendaItemModel).hasMany('sessions')?.find((session) => {
```

Reply:
Type `this`, TS is magic :heart_eyes: